platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/roster/RosterUtilsTest.java

```diff
@@ -17,29 +17,154 @@
+    }
+
+    @Test
+    @DisplayName("Test construct initial roster without software upgrade")
+    void testConstructInitialRosterWithoutSoftwareUpgrade() {
+        final SoftwareVersion version = mock(SoftwareVersion.class);
+        final ReservedSignedState initialState = mock(ReservedSignedState.class);
+        final SignedState state = mock(SignedState.class);
+        final MerkleRoot stateMerkleRoot = mock(MerkleRoot.class);
+        final WritableRosterStore rosterStore = mock(WritableRosterStore.class);
+        final PlatformStateAccessor platformState = mock(PlatformStateAccessor.class);
+
+        when(initialState.get()).thenReturn(state);
+        when(state.getState()).thenReturn(stateMerkleRoot);
+        when(stateMerkleRoot.getWritableRosterStore()).thenReturn(rosterStore);
+        when(state.getState().getWritableRosterStore()).thenReturn(rosterStore);
+        when(stateMerkleRoot.getReadablePlatformState()).thenReturn(platformState);
+        when(state.getState().getWritablePlatformState()).thenReturn(platformState);
+        when(rosterStore.getCandidateRoster()).thenReturn(null);
+
+        assertNotNull(RosterUtils.constructInitialRoster(version, initialState, addressBook));
+    }
+
+    @Test
+    @DisplayName(
+            "Test construct initial roster without software upgrade but with an active roster present in the state")
+    void testConstructInitialRosterWithoutSoftwareUpgradeWithActiveRoster() {
+        final SoftwareVersion version = mock(SoftwareVersion.class);
+        final ReservedSignedState initialState = mock(ReservedSignedState.class);
+        final SignedState state = mock(SignedState.class);
+        final MerkleRoot stateMerkleRoot = mock(MerkleRoot.class);
+        final WritableRosterStore rosterStore = mock(WritableRosterStore.class);
+        final Roster activeRoster = mock(Roster.class);
+        final PlatformStateAccessor platformState = mock(PlatformStateAccessor.class);
+
+        when(initialState.get()).thenReturn(state);
+        when(state.getState()).thenReturn(stateMerkleRoot);
+        when(stateMerkleRoot.getWritableRosterStore()).thenReturn(rosterStore);
+        when(rosterStore.getActiveRoster()).thenReturn(activeRoster);
+        when(stateMerkleRoot.getWritableRosterStore()).thenReturn(rosterStore);
+        when(state.getState().getWritableRosterStore()).thenReturn(rosterStore);
+        when(stateMerkleRoot.getReadablePlatformState()).thenReturn(platformState);
+        when(state.getState().getWritablePlatformState()).thenReturn(platformState);
+
+        assertSame(activeRoster, RosterUtils.constructInitialRoster(version, initialState, addressBook));
+    }
+
+    @Test
+    void testCreateRoster() {
+        when(addressBook.getSize()).thenReturn(1);
+        when(addressBook.getNodeId(0)).thenReturn(nodeId);
+        when(addressBook.getAddress(nodeId)).thenReturn(address);
+
+        assertNotNull(RosterUtils.createRoster(addressBook));
```

Comment:
This is not a sufficient assertion. This test would also be better if it used an actual instance of AddressBook. It's very easy to create one with `RandomAddressBookBuilder`

Reply:
Good points on the tests, I've extended them.
But the logic here mostly tests for Sameness and Equality, i don't need an actual address book to verify that- one less randomness in my tests.